src/App.js

```diff
@@ -6,22 +6,24 @@ import {Route, Redirect , Switch} from "react-router-dom"
 import Register from "./Containers/Register/Register";
 import MainLayout from "./Components/Layout/MainLayout";
 import * as actions from "./Store/Action/Login.jsx";
-import {useDispatch,connect} from "react-redux";
+import {useDispatch,useSelector} from "react-redux";
 
 function App(props) {
   const dispatch = useDispatch()
+  const auth = useSelector(state => state);
 
   useEffect(()=>{
     dispatch(actions.authCheckState());
-  },[])
+  },[dispatch])
   let routes = (
     <Switch>
       <Route path="/Login" exact component={Login}/>,
       <Route path="/Register" exact component={Register}/>,
       <Redirect to="/" from="/Login"/>
     </Switch>
   );
-  if (props.isAuthenticated) {
+
+  if (auth.Login.token !== null) {
```

Comment:
You don't need to check like this `if (auth.Login.token !== null) `.
You can check condition like this `if (auth.Login.token) `